plugins/inputs/tacacs/tacacs.go

```diff
@@ -0,0 +1,164 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package tacacs
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/nwaples/tacplus"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Tacacs struct {
+	Servers         []string        `toml:"servers"`
+	Username        config.Secret   `toml:"username"`
+	Password        config.Secret   `toml:"password"`
+	Secret          config.Secret   `toml:"secret"`
+	RemAddr         string          `toml:"remaddr"`
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+	Log             telegraf.Logger `toml:"-"`
+}
+
+//go:embed sample.conf
+var sampleConfig string
+
+func (t *Tacacs) SampleConfig() string {
+	return sampleConfig
+}
+
+func (t *Tacacs) Init() error {
+	if len(t.Servers) == 0 {
+		t.Servers = []string{"127.0.0.1"}
+	}
+	return nil
+}
+
+func (t *Tacacs) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, server := range t.Servers {
+		wg.Add(1)
+		go func(server string) {
+			defer wg.Done()
+			acc.AddError(t.pollServer(acc, server))
+		}(server)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (t *Tacacs) pollServer(acc telegraf.Accumulator, server string) error {
+	// Create the fields for this metric
+	tags := map[string]string{"source": server}
+	fields := make(map[string]interface{})
+
+	secret, err := t.Secret.Get()
+	if err != nil {
+		return fmt.Errorf("getting secret failed: %v", err)
+	}
+	defer config.ReleaseSecret(secret)
+
+	username, err := t.Username.Get()
+	if err != nil {
+		return fmt.Errorf("getting username failed: %v", err)
+	}
+	defer config.ReleaseSecret(username)
+
+	password, err := t.Password.Get()
+	if err != nil {
+		return fmt.Errorf("getting password failed: %v", err)
+	}
+	defer config.ReleaseSecret(password)
+
+	// Create the tacacs client
+	client := &tacplus.Client{
+		Addr: server,
+		ConnConfig: tacplus.ConnConfig{
+			Secret: secret,
+		},
+	}
+
+	// Initialize the AuthStart data
+	testAuthStart := &tacplus.AuthenStart{
+		Action:        tacplus.AuthenActionLogin,
+		AuthenType:    tacplus.AuthenTypeASCII,
+		AuthenService: tacplus.AuthenServiceLogin,
+		PrivLvl:       1,
+		Port:          "heartbeat",
+		RemAddr:       t.RemAddr,
+	}
+
+	// send the start request, the reply should be AuthenStatusGetUser
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(t.ResponseTimeout))
+	defer cancel()
+	startTime := time.Now()
+	reply, session, err := client.SendAuthenStart(ctx, testAuthStart)
+	if err != nil {
+		t.Log.Warnf("error on new tacacs authentication start request to %s : %s", server, err)
+		fields["responsetime"] = time.Duration(t.ResponseTimeout).Seconds()
+		fields["responsetime_ms"] = time.Duration(t.ResponseTimeout).Milliseconds()
+		acc.AddFields("tacacs", fields, tags)
+		return nil
```

Comment:
I think we should do the same as for the radius plugin. Remove one of the redundant `responsetime` fields and add some information to the metric indicating the "error-code"/"status".